### DISCUSSION TITLE
Configuring response headers

### DISCUSSION THREAD

**Discussion by kellytk** [tokio-rs/axum#180] [Q&A] [answered]
HTML response [hardcodes a `text/html` content type](https://github.com/tokio-rs/axum/blob/main/src/response/mod.rs#L414), however XHTML5 requires[0]:

> When a document is transmitted with an XML MIME type, such as application/xhtml+xml, then it is treated as an XML document by Web browsers, to be parsed by an XML processor.

Where please is it documented how that should be configured with `axum`?  If it isn't, modify the response's header in the handler, mostly duplicate [HTML response](https://github.com/tokio-rs/axum/blob/main/src/response/mod.rs#L398) to an XHTML response pull request, other?

[0] https://www.w3.org/TR/html51/introduction.html 1.6. HTML vs XHTML

**Comment by davidpdrsn** [comment, marked as answer]
I would recommend you duplicate the `Html` response and customize it to your needs.

You could also write a middleware that looks for responses with an html content type and then change it.
